compiler/compile/VirtualGuard.cpp

```diff
@@ -213,6 +213,60 @@ TR_VirtualGuard::createMethodGuard
    return createMethodGuardWithReceiver (kind, comp, calleeIndex, callNode, destination, calleeSymbol, thisClass, callNode->getSecondChild());
    }
 
+/*
+ * Shape of a breakpoint guard:
+ * iflcmpeq
+ *    land
+ *       aloadi <ConstantPool shadow>
+ *          aconst J9Method
+ *       isBreakpointedBit 
+ *    isBreakpointedBit 
+ */
+TR::Node*
+TR_VirtualGuard::createBreakpointGuard
+(TR::Compilation * comp, int16_t calleeIndex,
+ TR::Node* callNode, TR::TreeTop * destination, TR::ResolvedMethodSymbol * calleeSymbol)
+   {
+#ifdef J9_PROJECT_SPECIFIC
+   TR::SymbolReferenceTable * symRefTab = comp->getSymRefTab();
+   TR::SymbolReference * addressSymRef = symRefTab->findOrCreateJ9MethodConstantPoolFieldSymbolRef(offsetof(struct J9Method, constantPool));
+   TR::Node * aconstNode = TR::Node::aconst(callNode, (uintptrj_t)calleeSymbol->getResolvedMethod()->getPersistentIdentifier());
+   TR::Node * constantPool = TR::Node::createWithSymRef(TR::aloadi, 1, 1, aconstNode, addressSymRef);
+   aconstNode->setIsMethodPointerConstant(true);
+   aconstNode->setInlinedSiteIndex(calleeIndex);
+   aconstNode->setByteCodeIndex(0);
+
+   TR::Node *guard = NULL;
+   if (TR::Compiler->target.is64Bit())
+      {
+      TR::Node * flagBit = TR::Node::create(callNode, TR::lconst, 0, 0);
+      flagBit->setLongInt(comp->fej9()->offsetOfMethodIsBreakpointedBit());
+
+      guard =  TR::Node::createif(TR::iflcmpeq,
+                        TR::Node::create(TR::land, 2, constantPool, flagBit),
+               flagBit,
+               destination);
+      }
+   else
+      {
+      TR::Node * flagBit = TR::Node::create(callNode, TR::iconst, 0, comp->fej9()->offsetOfMethodIsBreakpointedBit());
+
+      guard =  TR::Node::createif(TR::ificmpne,
```

Comment:
we agree the l needs to change to an i but one case is iflcmpeq and in the other it is ificmpne - why do you flip the sense of the test - that is not what you want  don't think they should both be eq or ne.